### PR TITLE
OCPBUGS-26067: Backport CEL IP and CIDR validations

### PR DESCRIPTION
**Pull request by JoelSpeed**
A recent addition upstream includes new CEL validations for IPs and CIDRs. We have a number of APIs in tech preview in 4.15 that intend to be promoted to GA in 4.16, that could benefit from these IP address validations.

Our current approach is to use regex which does not provide a good experience when an error occurs, the whole regex is spit out.

The intention here is to backport the library change, which can be dropped when rebasing to 1.30. And the also change the introduced version to 1.28 (which also means this needs backporting to 4.15), so that, we ship the library in 4.15, but only start using it in 4.16.

Upstream, they were only introduced in 1.30 which means they wouldn't be usable until 1.31.

This allows us to improve our API validation from OpenShift 4.16 onwards, instead of OpenShift 4.18 onwards.

Previously discussed with @deads2k 
/assign @deads2k

### CODE REVIEW

**Comment by JoelSpeed** [comment]
/hold

I've rebase this to make sure it's up to date with the 1.29 changes, will be ready for merge once #1815 merges and I've rebased again

Will also be bringing up for discussion in the new year

**Comment by JoelSpeed** [comment]
/hold cancel

The 1.29 rebase has merged and I've got this rebased, assuming tests pass I think this could be merged

**Comment by JoelSpeed** [comment]
Unit failure looks unrelated
/test unit

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
